### PR TITLE
IGNITE-22020 : DumpReder misses 'keepRaw' parameter

### PR DESCRIPTION
**Pull request by Vladsz83**

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/AbstractSnapshotSelfTest.java [hunk 1]
```diff
@@ -1079,7 +1079,7 @@ public Account(int id, int balance) {
     }
 
     /** */
-    protected static class BlockingExecutor implements Executor {
+    public static class BlockingExecutor implements Executor {
```
Comment (timoninmaxim) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/AbstractSnapshotSelfTest.java:1082]: It's a bad practice make the internal static classes public. Let's revert these changes, maybe it's better to inherit the test and invoke `doSnapshotCancellationTest` method?

**Review by timoninmaxim — CHANGES_REQUESTED** [review]

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/dump/IgniteCacheDumpSelf2Test.java [hunk 2]
```diff
@@ -170,6 +174,85 @@ public class IgniteCacheDumpSelf2Test extends GridCommonAbstractTest {
...
+
+            @Override public void onCacheConfigs(Iterator<StoredCacheData> caches) {
+                // No-op.
+            }
+
+            @Override public void onPartition(int grp, int part, Iterator<DumpEntry> data) {
+                data.forEachRemaining(de -> {
+                    if (keepRaw.get()) {
+                        assertTrue(de.key() instanceof KeyCacheObject);
+                        assertTrue(de.value() instanceof CacheObject);
```
[the remark this answers is not in the document]
Comment (Vladsz83) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/dump/IgniteCacheDumpSelf2Test.java:215]: We say that we return a `CacheObject` in the javadocs if `keepRaw` is enabled. We check the `CacheObject`. It is.

**Review by timoninmaxim — APPROVED** [review]
